### PR TITLE
DivisiveCluster: "model_" attribute for all nodes & fcluster option

### PR DESCRIPTION
**Pull request by tliu68**
#### Reference Issues/PRs

address feedback on PR #398 

#### What does this implement/fix? Explain your changes.

1. add a `.model_` attribute to nodes that were not fitted
2. add an `fcluster` option where estimated/predicted labels can be renumbered so that the indices of the clusters match those of the corresponding nodes in the dendrogram, and the renumbered labels become column-stacked flat clusterings

### CODE REVIEW

**Comment by tliu68** [comment]
@bdpedigo I didn't include the tutorial here in the spirit of small PRs but please let me know if you think I should. Thanks!

**Comment by bdpedigo** [comment]
> @bdpedigo I didn't include the tutorial here in the spirit of small PRs but please let me know if you think I should. Thanks!

nope you are correct :)

File: graspologic/cluster/divisive_cluster.py [hunk 1]
```diff
@@ -137,20 +149,28 @@ def fit(self, X):
...
         """
         Fits clustering models to the data as well as resulting clusters
         and using fitted models to predict a hierarchy of labels
 
         Parameters
         ----------
         X : array-like, shape (n_samples, n_features)
+        fcluster: bool, default=False
+            if True, returned labels will be a hierarchy of flat clusterings,
+            one for each level
```
Comment (bdpedigo) [inline graspologic/cluster/divisive_cluster.py:162]: just documenting from our conversation wednesday - the wording here confused me in terms of what this is actually doing

File: graspologic/cluster/divisive_cluster.py [hunk 2]
```diff
@@ -238,26 +265,68 @@ def _fit(self, X):
...
+                        if parent_model.means_init is not None:
+                            model.update(
+                                {
+                                    "means_init": parent_model.means_init[cluster_indx],
+                                    "precisions_init": parent_model.precisions_init[
+                                        cluster_indx
+                                    ],
+                                    "weights_init": np.array([1]),
+                                }
+                            )
```
Comment (bdpedigo) [inline graspologic/cluster/divisive_cluster.py:288]: also from our conversation wednesday - I felt like we should either decide not to have this approach or to create a true GaussianMixture object for each node in the clustering tree. I don't feel super strongly which and happy to discuss more, I did feel that what we have here right now is kinda halfway inbetween

Comment (bdpedigo) [reply graspologic/cluster/divisive_cluster.py]: i am ready to merge besides this point - I still feel like what is in this PR is a half measure between an actual fitted GM model and nothing, so right now I'm thinking we just omit in this PR until we decide something else. 

I think the .model attribute is only useful if it has a fitted mean and covariance, is why I'm still uncertain about this. currently im not sure why we'd need the mean_init and precision_init, for instance

File: graspologic/cluster/divisive_cluster.py [hunk 3]
```diff
@@ -238,26 +268,86 @@ def _fit(self, X):
                             (labels, np.zeros((len(X), 1), dtype=int))
                         )
                     labels[inds, 1 : child_labels.shape[1] + 1] = child_labels
+                else:
+                    # make a "GaussianMixture" model for clusters
+                    # that were not fitted
+                    if self.cluster_method == "gmm":
+                        cluster_idx = len(dc.parent.children) - 1
+                        parent_model = dc.parent.model_
+                        model = GaussianMixture()
```
Comment (tliu68) [inline graspologic/cluster/divisive_cluster.py:277]: @bdpedigo I made some simple changes below. Is it close to what we want? If not, I'll just take this part out for now as you suggested.

Comment (bdpedigo) [reply graspologic/cluster/divisive_cluster.py:277]: this seems more in line with what i was thinking, I'm just going to merge for now and we can improve later if need be. have you tested this out at all?

**Review by bdpedigo — APPROVED** [review]

File: graspologic/cluster/divisive_cluster.py (hunk 3 above)
Comment (bdpedigo) [reply graspologic/cluster/divisive_cluster.py:277]: possibly something to mention/test in a tutorial but we can talk more later

**Pull request merged**